omod/src/main/webapp/pages/allergies.gsp

```diff
@@ -84,7 +84,7 @@ ${ ui.includeFragment("uicommons", "infoAndErrorMessage")}
 	                <td>
 	                	<i class="icon-pencil edit-action" title="${ ui.message("coreapps.edit") }"
 	                       onclick="location.href='${ ui.pageLink("allergyui", "allergy", [allergyId:allergy.id, patientId: patient.id, returnUrl: returnUrl]) }'"></i>
-	                    <i class="icon-remove delete-action" title="${ ui.message("coreapps.delete") }" onclick="removeAllergy('${ allergy.allergen }', ${ allergy.id})"></i>
+	                    <i class="icon-remove delete-action" title="${ ui.message("coreapps.delete") }" onclick="removeAllergy('${ allergy.allergen.escapeJavascript }', ${ allergy.id})"></i>
```

Comment:
@jcantu1988 that is problematic when the module is running on platform 2.0 because then there is no allergyapi module. Can you think of another solution that does not touch the allergyapi module?